### PR TITLE
go.mod changes: less replace

### PR DESCRIPTION
**Pull request by vdemeester**
# Changes

This PR removes as much `replace` as possible as they won't "transpire" in modules depending on `tektoncd/pipeline`. I think `replace` should be used as last resort (a bit like `[[override]]` was to be used using `dep`). There is two commits in there:

- go.mod: remove replace on tektoncd/plumbing
- <del>go.mod: remove replace for kubernetes</del>

<del>For kubernetes, putting kubernetes in `replace` means we don't tell
dependency which version we are depending too, which could lead with
compilation failure (as they would get higher version).</del>

For plumbing, this allows to do a `go get -u github.com/tektoncd/plumbing` and actually depending on what was updated.

NB: we need to had a check on the `vendor` folder, to make sure the `vendor` folder is in sync with the `go.mod`.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md) for more details._

Double check this list of stuff that's easy to miss:

- If you are adding [a new binary/image to the `cmd` dir](../cmd), please update
  [the release Task](../tekton/publish.yaml) to build and release this image.

## Reviewer Notes

If [API changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md) are included, [additive changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#additive-changes) must be approved by at least two [OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS) and [backwards incompatible changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-incompatible-changes) must be approved by [more than 50% of the OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS), and they must first be added [in a backwards compatible way](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-compatible-changes-first).

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -40,62 +39,33 @@ require (
...
-	k8s.io/kube-aggregator => k8s.io/kube-aggregator v0.0.0-20191004103911-2797d0dcf14b
-	k8s.io/kube-controller-manager => k8s.io/kube-controller-manager v0.0.0-20191016015407-72acd948ffff
-	k8s.io/kube-proxy => k8s.io/kube-proxy v0.0.0-20191016015246-999188f3eff6
-	k8s.io/kube-scheduler => k8s.io/kube-scheduler v0.0.0-20191016015341-7be46aeada42
-	k8s.io/kubelet => k8s.io/kubelet v0.0.0-20191016015314-e7fc4f69fc2c
-	k8s.io/kubernetes => k8s.io/kubernetes v1.13.12
-	k8s.io/metrics => k8s.io/metrics v0.0.0-20191004105814-56635b1b5a0c
 )
-
-// Local overrides for testing. Remove before submitting.
```
Comment (vdemeester) [inline go.mod]: @wlynch we forgot to remove before merging :sweat_smile:

**Comment by vdemeester** [comment]
ok nevermind, the `replace` for `kubernetes` deps seem necessary… because we have an import on `k8s.io/kubernetes` that.. has `v0.0.0`… oh my.. I don't like `go.mod` + kubernetes already :sweat_smile:

**Comment by chmouel** [comment]
/lgtm

Thanks 👍🏼🔥

**Pull request merged**